### PR TITLE
Deprecate the old cbr0 and flannel networking modes

### LINKED ISSUE #19766: kubelet sets veth hairpin mode regardless of network plugin
From another issue, @shettyg said:

> I connect my pod interface to a Open vSwitch bridge. Since this code does not find a peer interface on a linux bridge,
> it just enabled hairpin mode in every linux bridge on my host. This completely breaks my host where I do not want hair
> pin mode enabled. It took me a good 2 days to figure out why things break in my setup. This feels like a very bad way
> to solve problems.

DockerManager.SyncPod() calls the network Plugin and then sets hairpin mode unilaterally.  This is wrong. We should either pass the `configureCBR0` bool down to here or else we should get rid of the default no-op plugin and use the non-existence of a plugin as a flag to control this.

This should be a relatively easy one for people who want to contribute but don't know where to start.

### PR DESCRIPTION
**Pull request by freehan**
fixes: #19766

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31197)

### CODE REVIEW

**Comment by yujuhong** [comment]
LGTM. I'll leave it to @freehan to see if he wants to take my LGTM or add more reviews.

File: cmd/kubelet/app/options/options.go [hunk 1]
```diff
@@ -168,6 +168,7 @@ func (s *KubeletServer) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.RktStage1Image, "rkt-stage1-image", s.RktStage1Image, "image to use as stage1. Local paths and http/https URLs are supported. If empty, the 'stage1.aci' in the same directory as '--rkt-path' will be used.")
 	fs.MarkDeprecated("rkt-stage1-image", "Will be removed in a future version. The default stage1 image will be specified by the rkt configurations, see https://github.com/coreos/rkt/blob/master/Documentation/configuration.md for more details.")
 	fs.BoolVar(&s.ConfigureCBR0, "configure-cbr0", s.ConfigureCBR0, "If true, kubelet will configure cbr0 based on Node.Spec.PodCIDR.")
+	fs.MarkDeprecated("configure-cbr0", "Will be removed in a future version. Please use kubenet or other network plugins.")
```
Comment (luxas) [inline cmd/kubelet/app/options/options.go]: Please mark experimental flannel overlay as deprecated also

**Comment by luxas** [comment]
@freehan please rebase

**Comment by luxas** [comment]
I took the liberty to edit the PR title for a release note.
Does it look good to you?

**Comment by luxas** [comment]
This LGTM

**Comment by freehan** [comment]
Thanks. The title is much better

**Comment by yujuhong** [comment]
LGTM. Thanks!

**Pull request merged**